azure-iot-device/azure/iot/device/common/pipeline/operation_flow.py

```diff
@@ -92,13 +92,21 @@ def pass_op_to_next_stage(stage, op):
 
 
 @pipeline_thread.runs_on_pipeline_thread
-def complete_op(stage, op):
+def complete_op(stage, op, error=None):
     """
     Helper function to complete an operation by calling its callback function thus
     returning the result of the operation back up the pipeline.  This is perferred to
     calling the operation's callback directly as it provides several layers of protection
     (such as a try/except wrapper) which are strongly advised.
     """
+    if error:
+        if op.error:
```

Comment:
>if op.error [](start = 8, length = 11)

i agree that we need to narrow this down to a single way of doing things, but, I suggest `if op.error and op.error not == error:` to make the warning a little more explicit.